pkg/cosesign1/Makefile

```diff
@@ -23,21 +23,31 @@
 # will print the new fingerprint of the intermediate cert as part of the did:x509 generated
 #
 
+
+# note test-fail is expected to fail
+
+all: chain.pem cose test-fail test-pass
+
 cose: infra.rego.cose
 
 # from these media types have to match containerd. The also need to change and the security policy one ought to be x-ms-ccepolicy-frag
 #     fragment atrifact type = application/x-ms-ccepolicy-frag
-#     fragment media type = application/cose_x509+rego
+#     fragment media type = application/cose-x509+rego
 
 # Use a local linux build of the tool for the purposes of this Makefile - ie assume using in wsl.
 # Usually sign1util.exe is a windows exe in /mnt/c/ContainerPlat aka c:\ContainerPlat but that is not certain.
 
 sign1util: ../../cmd/sign1util/main.go *.go
 	go build ../../cmd/sign1util
 
-infra.rego.cose: infra.rego.base64 chain.pem leaf.private.pem sign1util
-	./sign1util create -algo ES384 -chain chain.pem -claims infra.rego.base64 -key leaf.private.pem -out $@ -issuer TestIssuer -feed TestFeed -salt zero
-	./sign1util check -in $@
+#infra.rego.cose: infra.rego chain.pem leaf.private.pem sign1util
+#	./sign1util create -algo ES384 -chain chain.pem -claims infra.rego -key leaf.private.pem -out $@ -issuer TestIssuer -feed TestFeed -salt zero
+#	./sign1util check -in $@
+
+%.rego.cose: %.rego chain.pem leaf.private.pem sign1util
+	./sign1util create -algo ES384 -chain chain.pem -claims $< -key leaf.private.pem -out $@ -salt zero \
+	-feed acceuroperegistry.azurecr.io/infra:latest -content-type application/unknown+rego \
+ 	-issuer TestIssuer -feed TestFeed
```

Comment:
can you actually define `FEED` and use the actual issuer DID?
```
...
-feed $(FEED) \
-issuer $(ISSUER_DID)
```
It took me some time to figure out that the issuer/feed were wrong in the fragment that I pushed.

Reply:
> It makes a fresh one every time you do a make clean. Otherwise they get stale, and the secrets police get on your case.

What I mean is you're doing this below:
```
ISSUER_DID = $(shell ./sign1util did-x509 -chain chain.pem -policy cn)
```

Can't you do the same (or reuse the value) for the `%.rego.cose` target? The `-feed` could be a makefile parameter or something, don't you think?
```
%.rego.cose: %.rego chain.pem leaf.private.pem sign1util
    ISSUER_DID=$(shell ./sign1tuil did-x509 -chain chain.pem -policy cn)
    ./sign1util create -algo ...\
        -feed $(FEED)
        -issuer $(ISSUER_DID)
```


So, this way, I would be able to create any fragment (given that I've put correct `infra.rego`):
```
make FEED=myregistry.azurecr.io/my/fragment:latest
```

Reply:
also I just noticed, it seems like you're passing `-feed` twice to `sign1util` here.